boa_engine/src/vm/mod.rs

```diff
@@ -1521,7 +1521,17 @@ impl Context {
                     let function = function_object
                         .as_function()
                         .expect("must be function object");
-                    function.get_home_object().cloned()
+                    let mut home_object = function.get_home_object().cloned();
+
+                    if home_object == None {
```

Comment:
Functional style is usually preferred in these cases:
```suggestion
                    if home_object.is_none() {
```